lib/model.js

```diff
@@ -1857,21 +1857,13 @@ class Model {
       countOptions.attributes = undefined;
     }
 
-    const countQuery = this.count(countOptions);
-    const findQuery = this.findAll(options);
-
-    return countQuery.then(count => {
-      if (count === 0) {
-        return {
-          count: 0,
-          rows: []
-        };
-      }
-      return findQuery.then(results => ({
```

Comment:
`findQuery` should reject with error promise, which user then should be able to handle in their code. I suspect it could rejecting synchronously?

Reply:
User calls findAndCount and he able to handle result in then/catch (he don't should be able handle countQuery and findQuery separately). And my code allows it to do. I do not understand what the problem is. could you explain in more detail?